packages/language/src/pli.langium

```diff
@@ -628,10 +652,15 @@ ExpExpression infers Expression: PrimaryExpression ({infer ExpExpression.left=cu
 
 PrimaryExpression infers Expression: 
     Literal
-    | '(' Expression ')'
+    | ParenthesizedExpression
     | UnaryExpression
     | LocatorCall;
 
+// Performs special handling as a recursive descent parser otherwise cannot handle the 
+// special data-list type-3-DO loop
+// See also language reference pp. 301
+ParenthesizedExpression infers Expression: {infer Parenthesis} '(' value=Expression ('DO' do=DoType3<false>)? ')' ({infer Literal.multiplier=current} value=LiteralValue)?;
```

Comment:
Not sure about this part w/ the syntactic predicate. Parse results look okay, but I'm not sure how well we test this case, probably need to cover that in more detail coming up.